pkg/mock/fileUtils.go

```diff
@@ -13,9 +13,12 @@ import (
 
 var dirContent []byte
 
+const keyContent = "content"
+const keyMode = "mode"
+
 // FilesMock implements the functions from piperutils.Files with an in-memory file system.
 type FilesMock struct {
-	files        map[string]*[]byte
+	files        map[string]map[string]interface{}
```

Comment:
Why not define a private, type-safe struct here that has the content byte-pointer and permissions info and then map to that. What is the benefit of mapping to interface{}?

Reply:
no benefit, adjusted accordingly